main/main.go

```diff
@@ -0,0 +1,260 @@
+		allIstioCVEs = append(allIstioCVEs, istioCVEs...)
+	}
+	log.Infof("total %d k8s and %d istio CVEs found", len(allK8sCVEs), len(allIstioCVEs))
+	return allK8sCVEs, allIstioCVEs, nil
+}
+
+func filterCVEs(cves *utils.NvdCVEs, project string) []utils.NvdCVEEntry {
+	var cveEntries []utils.NvdCVEEntry
+	for _, cve := range cves.Entries {
+		if appliesToProject(cve, project) {
+			cveEntries = append(cveEntries, cve)
+		}
+	}
+	return cveEntries
+}
+
+func appliesToProject(cve utils.NvdCVEEntry, project string) bool {
+	for _, node := range cve.Configurations.Nodes {
+		for _, cpeMatch := range node.CPEMatch {
+			if strings.Contains(cpeMatch.CPE23Uri, fmt.Sprintf("%s:%s", project, project)) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func getDataFeeds(path string) (map[string]string, error) {
+	dataFeedReaders := make(map[string]string)
+
+	// Get hashes for these feeds.
+	for y := 2014; y <= time.Now().Year(); y++ {
+		dataFeedName := strconv.Itoa(y)
+		if err := validateURLMeta(utils.GetDataFeedMetaURL(dataFeedName)); err != nil {
+			log.WithError(err).WithField(logDataFeedName, dataFeedName).Warning("could not get NVD data feed hash")
+			continue
+		}
+
+		fileName := filepath.Join(path, fmt.Sprintf("%s%s", dataFeedName, cveFileExt))
+		if err := downloadFeed(dataFeedName, fileName); err != nil {
+			return dataFeedReaders, err
+		}
+		dataFeedReaders[dataFeedName] = fileName
+	}
+	return dataFeedReaders, nil
+}
+
+func downloadFeed(dataFeedName, fileName string) error {
+	r, err := utils.RunHTTPGet(utils.GetDataFeedURL(dataFeedName))
+	if err != nil {
+		log.WithError(err).WithField(logDataFeedName, dataFeedName).Error("could not download NVD data feed")
+		return err
+	}
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		log.WithFields(log.Fields{"StatusCode": r.StatusCode, "Status": r.Status, "DataFeedName": dataFeedName}).Error("failed to download NVD data feed")
+		buf, err := utils.ReadNBytesFromResponse(r, 1024)
+		if err != nil {
+			return errors.Wrapf(err, "failed to read response body, status code: %d, status: %s", r.StatusCode, r.Status)
+		}
+		return fmt.Errorf("failed to download NVD data feed. status code: %d, status: %s, response body: %s", r.StatusCode, r.Status, string(buf))
+	}
+
+	err = utils.StoreHTTPResponseToFile(r, fileName, dataFeedName)
+	if err != nil {
+		log.WithFields(log.Fields{"DataFeedName": dataFeedName}).Errorf("failed to store gzip response to file")
+		return err
+	}
+
+	return nil
+}
+
+func validateURLMeta(metaURL string) error {
+	r, err := utils.RunHTTPGet(metaURL)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	if r.StatusCode != http.StatusOK {
+		log.WithFields(log.Fields{"StatusCode": r.StatusCode, "Status": r.Status}).Error("failed to get NVD data feed meta")
+		buf, err := utils.ReadNBytesFromResponse(r, 1024)
+		if err != nil {
+			return errors.Wrapf(err, "failed to read response body, status code: %d, status: %s", r.StatusCode, r.Status)
```

Comment:
Not a huge deal, but it's better to say "failed to get NVD data feed meta... Additionally, there was an error reading the response body" (since the primary cause of the error is the status code not matching, and we're only reading the response body to augment the message. (Same applies above.)